src/root.zig

```diff
@@ -0,0 +1,451 @@
+const std = @import("std");
+const enrlib = @import("enr");
+
+pub const ENR = enrlib.ENR;
+pub const EncodedENR = enrlib.EncodedENR;
+pub const SignableENR = enrlib.SignableENR;
+
+/// ENR file I/O error set
+pub const EnrFileError = error{
+    /// The ENR file is too large to process
+    EnrFileTooLarge,
+};
+
+/// Loads ENR from the given file path
+pub fn loadEnrFromDisk(enr: *ENR, file_path: []const u8) !void {
+    const enr_file = try std.fs.cwd().openFile(file_path, .{});
+    defer enr_file.close();
+
+    const file_size = try enr_file.getEndPos();
+    if (file_size > enrlib.max_enr_size) return EnrFileError.EnrFileTooLarge;
+
+    var enr_txt: [enrlib.max_enr_size]u8 = undefined;
+    _ = try enr_file.readAll(enr_txt[0..file_size]);
+
+    try ENR.decodeTxtInto(enr, enr_txt[0..file_size]);
+}
+
+/// Loads EncodedENR from the given file path
+pub fn loadEncodedEnrFromDisk(encoded_enr: *EncodedENR, file_path: []const u8) !void {
+    const enr_file = try std.fs.cwd().openFile(file_path, .{});
+    defer enr_file.close();
+
+    const file_size = try enr_file.getEndPos();
+    if (file_size > enrlib.max_enr_size) return EnrFileError.EnrFileTooLarge;
```

Comment:
it would make sense to have only one functions that reads data from disk, and then decide how you process that data in `loadEncodedEnrFromDisk` and `loadEnrFromDisk`